rdr_service/resource/generators/_base.py

```diff
@@ -286,26 +289,44 @@ def _lookup_code_value(self, code_id, session):
         :param session: DAO session object
         :return: string
         """
+        global _CODEBOOK_CODE_VALUES
         if code_id is None:
             return None
-        result = session.query(Code.value).filter(Code.codeId == int(code_id)).first()
-        if not result:
-            return None
-        return result.value
 
-    def _lookup_code_id(self, code, ro_session):
+        if _CODEBOOK_CODE_VALUES is None:
+            results = session.query(Code.codeId, Code.value).all()
+            if not results:
+                return None
+            _CODEBOOK_CODE_VALUES = dict()
+            for row in results:
+                _CODEBOOK_CODE_VALUES[row.codeId] = row.value
```

Comment:
It may be a rare issue, if it ever happens at all. But it looks like this isn't thread-safe and two threads could attempt to generate the `_CODEBOOK_CODE_VALUES` dict at the same time.

Reply:
That is a possibility, but even if two threads entered the block to set a global value at the same time, there would be no deadlock because of the Python Global Interpreter Lock and which ever one finished last would set the var to a dict successfully and the first dict would be discarded.  This change broke the unittests, so I'll look in to making some changes though.  A deadlock could be an issue in a different programming language that has true multi-threading.

Reply:
Additionally, both threads would create and set the exact same data here, which is different than say the logging code which needs to capture logging data per thread.